log ViewAndMutationMeta to trace_structured

bdhirsh opened the pull request:
I ended up bundling it into the existing tlparse logs for the AOT forward graph, since it looked like registering it as a separate artifact requires changes to tlparse itself (maybe that is wrong though?)

Example new fw AOT graph tlparse output for the below code: https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/.tmp70zKiO/0_0_0/aot_forward_graph_2.txt

```
import torch


@torch.compile
def f(x):
    out1 = torch.view_as_complex(x)
    out2 = torch.view_as_complex(x)
    return out1, out2, x * 2



x_ = torch.randn(4, 2, requires_grad=True, dtype=torch.float64)
out = f(x_)
```


Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #131481
* #133044
* #137921
* __->__ #133784



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

bdhirsh committed:
log ViewAndMutationMeta to trace_structured

[ghstack-poisoned]

ezyang commented on torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:
```diff
@@ -210,11 +210,21 @@ def _map_assigned_buffer_to_proxy(_mod, name, buffer):
-            payload_fn=lambda: fw_module.print_readable(
-                print_output=False, include_stride=True, include_device=True
-            ),
+            payload_fn=lambda: fw_metadata_str,
```
This doesn't do what you want, payload_fn=fw_metadata_str

Have you inspected the output tlparse yet?

ezyang replied:
Oh it looks like you didn't exercise this one

bdhirsh replied:
whoops... thank you

ezyang commented on torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:
```diff
@@ -210,11 +210,21 @@ def _map_assigned_buffer_to_proxy(_mod, name, buffer):
+                        print_output=False, include_stride=True, include_device=True
+                    ),
+                ]
+            )
```
Why put these together? Just output them as separate log entries. If you don't want to have to modify tlparse cli, dump the fw metadata as "artifact"

bdhirsh replied:
you're completely right - I didn't realize that dumping using "artifact" allows us to add new artifacts without needing to change the tlparse cli. updated

ezyang commented:
nit: I think the dumped structures would be substantially easier to understand if they were indented. There are two ways you could do this: dump as JSON, or use torchgen's dump dataclass with indentation utility

bdhirsh committed:
Update on "log ViewAndMutationMeta to trace_structured"

I ended up bundling it into the existing tlparse logs for the AOT forward graph, since it looked like registering it as a separate artifact requires changes to tlparse itself (maybe that is wrong though?)

Example new fw AOT graph tlparse output for the below code: https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/.tmp70zKiO/0_0_0/aot_forward_graph_2.txt

```
import torch


torch.compile
def f(x):
    out1 = torch.view_as_complex(x)
    out2 = torch.view_as_complex(x)
    return out1, out2, x * 2



x_ = torch.randn(4, 2, requires_grad=True, dtype=torch.float64)
out = f(x_)
```





[ghstack-poisoned]

bdhirsh committed:
Update on "log ViewAndMutationMeta to trace_structured"

I ended up bundling it into the existing tlparse logs for the AOT forward graph, since it looked like registering it as a separate artifact requires changes to tlparse itself (maybe that is wrong though?)

Example new fw AOT graph tlparse output for the below code: https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/.tmp70zKiO/0_0_0/aot_forward_graph_2.txt

```
import torch


torch.compile
def f(x):
    out1 = torch.view_as_complex(x)
    out2 = torch.view_as_complex(x)
    return out1, out2, x * 2



x_ = torch.randn(4, 2, requires_grad=True, dtype=torch.float64)
out = f(x_)
```





[ghstack-poisoned]

bdhirsh commented:
Just updated to use `dataclass_repr()` for formatting and it is much nicer https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/.tmp2lEs4q/0_0_1/aot_forward_graph_fw_metadata_2.txt

bdhirsh committed:
Update on "log ViewAndMutationMeta to trace_structured"

I ended up bundling it into the existing tlparse logs for the AOT forward graph, since it looked like registering it as a separate artifact requires changes to tlparse itself (maybe that is wrong though?)

Example new fw AOT graph tlparse output for the below code: https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/.tmp70zKiO/0_0_0/aot_forward_graph_2.txt

```
import torch


torch.compile
def f(x):
    out1 = torch.view_as_complex(x)
    out2 = torch.view_as_complex(x)
    return out1, out2, x * 2



x_ = torch.randn(4, 2, requires_grad=True, dtype=torch.float64)
out = f(x_)
```





[ghstack-poisoned]

bdhirsh committed:
Update on "log ViewAndMutationMeta to trace_structured"

I ended up bundling it into the existing tlparse logs for the AOT forward graph, since it looked like registering it as a separate artifact requires changes to tlparse itself (maybe that is wrong though?)

Example new fw AOT graph tlparse output for the below code: https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/.tmp70zKiO/0_0_0/aot_forward_graph_2.txt

```
import torch


torch.compile
def f(x):
    out1 = torch.view_as_complex(x)
    out2 = torch.view_as_complex(x)
    return out1, out2, x * 2



x_ = torch.randn(4, 2, requires_grad=True, dtype=torch.float64)
out = f(x_)
```





[ghstack-poisoned]

bdhirsh committed:
Update on "log ViewAndMutationMeta to trace_structured"

I ended up bundling it into the existing tlparse logs for the AOT forward graph, since it looked like registering it as a separate artifact requires changes to tlparse itself (maybe that is wrong though?)

Example new fw AOT graph tlparse output for the below code: https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/.tmp70zKiO/0_0_0/aot_forward_graph_2.txt

```
import torch


torch.compile
def f(x):
    out1 = torch.view_as_complex(x)
    out2 = torch.view_as_complex(x)
    return out1, out2, x * 2



x_ = torch.randn(4, 2, requires_grad=True, dtype=torch.float64)
out = f(x_)
```





[ghstack-poisoned]

bdhirsh committed:
Update on "log ViewAndMutationMeta to trace_structured"

I ended up bundling it into the existing tlparse logs for the AOT forward graph, since it looked like registering it as a separate artifact requires changes to tlparse itself (maybe that is wrong though?)

Example new fw AOT graph tlparse output for the below code: https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/.tmp70zKiO/0_0_0/aot_forward_graph_2.txt

```
import torch


torch.compile
def f(x):
    out1 = torch.view_as_complex(x)
    out2 = torch.view_as_complex(x)
    return out1, out2, x * 2



x_ = torch.randn(4, 2, requires_grad=True, dtype=torch.float64)
out = f(x_)
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

bdhirsh committed:
Update on "log ViewAndMutationMeta to trace_structured"

I ended up bundling it into the existing tlparse logs for the AOT forward graph, since it looked like registering it as a separate artifact requires changes to tlparse itself (maybe that is wrong though?)

Example new fw AOT graph tlparse output for the below code: https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/.tmp70zKiO/0_0_0/aot_forward_graph_2.txt

```
import torch


torch.compile
def f(x):
    out1 = torch.view_as_complex(x)
    out2 = torch.view_as_complex(x)
    return out1, out2, x * 2



x_ = torch.randn(4, 2, requires_grad=True, dtype=torch.float64)
out = f(x_)
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

bdhirsh committed:
Update on "log ViewAndMutationMeta to trace_structured"

I ended up bundling it into the existing tlparse logs for the AOT forward graph, since it looked like registering it as a separate artifact requires changes to tlparse itself (maybe that is wrong though?)

Example new fw AOT graph tlparse output for the below code: https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/.tmp70zKiO/0_0_0/aot_forward_graph_2.txt

```
import torch


torch.compile
def f(x):
    out1 = torch.view_as_complex(x)
    out2 = torch.view_as_complex(x)
    return out1, out2, x * 2



x_ = torch.randn(4, 2, requires_grad=True, dtype=torch.float64)
out = f(x_)
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

bdhirsh committed:
Update on "log ViewAndMutationMeta to trace_structured"

I ended up bundling it into the existing tlparse logs for the AOT forward graph, since it looked like registering it as a separate artifact requires changes to tlparse itself (maybe that is wrong though?)

Example new fw AOT graph tlparse output for the below code: https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/.tmp70zKiO/0_0_0/aot_forward_graph_2.txt

```
import torch


torch.compile
def f(x):
    out1 = torch.view_as_complex(x)
    out2 = torch.view_as_complex(x)
    return out1, out2, x * 2



x_ = torch.randn(4, 2, requires_grad=True, dtype=torch.float64)
out = f(x_)
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

bdhirsh committed:
Update on "log ViewAndMutationMeta to trace_structured"

I ended up bundling it into the existing tlparse logs for the AOT forward graph, since it looked like registering it as a separate artifact requires changes to tlparse itself (maybe that is wrong though?)

Example new fw AOT graph tlparse output for the below code: https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/.tmp70zKiO/0_0_0/aot_forward_graph_2.txt

```
import torch


torch.compile
def f(x):
    out1 = torch.view_as_complex(x)
    out2 = torch.view_as_complex(x)
    return out1, out2, x * 2



x_ = torch.randn(4, 2, requires_grad=True, dtype=torch.float64)
out = f(x_)
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

bdhirsh committed:
Update on "log ViewAndMutationMeta to trace_structured"

I ended up bundling it into the existing tlparse logs for the AOT forward graph, since it looked like registering it as a separate artifact requires changes to tlparse itself (maybe that is wrong though?)

Example new fw AOT graph tlparse output for the below code: https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/.tmp70zKiO/0_0_0/aot_forward_graph_2.txt

```
import torch


torch.compile
def f(x):
    out1 = torch.view_as_complex(x)
    out2 = torch.view_as_complex(x)
    return out1, out2, x * 2



x_ = torch.randn(4, 2, requires_grad=True, dtype=torch.float64)
out = f(x_)
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

bdhirsh commented:
@pytorchbot merge

bdhirsh committed:
Update on "log ViewAndMutationMeta to trace_structured"

I ended up bundling it into the existing tlparse logs for the AOT forward graph, since it looked like registering it as a separate artifact requires changes to tlparse itself (maybe that is wrong though?)

Example new fw AOT graph tlparse output for the below code: https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/.tmp70zKiO/0_0_0/aot_forward_graph_2.txt

```
import torch


torch.compile
def f(x):
    out1 = torch.view_as_complex(x)
    out2 = torch.view_as_complex(x)
    return out1, out2, x * 2



x_ = torch.randn(4, 2, requires_grad=True, dtype=torch.float64)
out = f(x_)
```





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

bdhirsh commented:
@pytorchbot merge

bdhirsh commented:
@pytorchbot merge
